File: RA1/JAVA/EjemploHerencia/Cat.java
package EjemploHerencia;

//la herencia se declara añadiendo la palabra "extends" de otra clase
// public class <subclass> extends <superclass>
public class Cat extends Animal{
    private String color;

    public Cat(String name, String color){
        super(name);//llamar al constructor de la clase superior, es decir, de clase Animal
        this.color = color;
    }

    //Polimorfismo: Inclusión
    @Override
    public void makeSound(){
        System.out.println(this.name + " maulla: Miau!!!");
    }

    //Polimorfismo: Paramétrico
    public void makeSound(String something){
        System.out.println(this.name + " maulla: " + something);
    }

    //Polimorfismo: Sobrecarga
    public void run(){
        System.out.println(this.name + " correr");
    }
}

File: RA1/PYTHON/ejemplo_herencia/cat.py
#from <archivo> import <elemento<class>>
from typing import overload
from animal import Animal

class Cat(Animal):
    def __init__(self, name, color):
        #opcion 1: 
        super().__init__(name)
        #opcion 2: Animal.__init__(name)
        self.color = color

    def make_sound(self):
        print(self.name + " hace MIAU...")

    # No se puede realizar polimorfismo paramétrico
    # def make_sound(self, something):
    #    print(self.name + "hace " + something)

    def run(self):
        print(self.name + " Corre")

    
    